platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/camera/CameraUpdateFactoryTest.kt

```diff
@@ -140,4 +143,50 @@ class CameraUpdateFactoryTest : BaseTest() {
       assertEquals("tilt should match:", 40.0, cameraPosition.tilt, 0.1)
     }
   }
+
+  @Test
+  fun withPadding_cameraInvalidated_paddingPersisting() {
+    rule.runOnUiThread {
+      val initialCameraPosition = mapboxMap.cameraPosition
+      val initialPoint = mapboxMap.projection.toScreenLocation(initialCameraPosition.target)
+
+      val bottomPadding = mapView.height / 4
+      val leftPadding = mapView.width / 4
+      val padding = doubleArrayOf(leftPadding.toDouble(), 0.0, 0.0, bottomPadding.toDouble())
+      mapboxMap.moveCamera(CameraUpdateFactory.paddingTo(leftPadding.toDouble(), 0.0, 0.0, bottomPadding.toDouble()))
+
+      Assert.assertArrayEquals(intArrayOf(leftPadding, 0, 0, bottomPadding), mapboxMap.padding)
+
+      val resultingCameraPosition = mapboxMap.cameraPosition
+      assertEquals(initialCameraPosition.target, resultingCameraPosition.target)
+      assertEquals(
+        PointF(initialPoint.x + leftPadding / 2, initialPoint.y - bottomPadding / 2),
+        mapboxMap.projection.toScreenLocation(resultingCameraPosition.target)
+      )
+      Assert.assertArrayEquals(padding, resultingCameraPosition.padding, 0.0001)
```

Comment:
we are comparing 2 camera positions by comparing all fields to eachother. Can we leverage CameraPosition#equals instead?

Reply:
In this case, we are comparing requested padding to a resulting one, not initial camera position to resulting camera position. We cannot compare the requested position to a resulting one because of no-op values like `-1` when the property doesn't need changing.